### PR TITLE
feat: 1655 Price search page without functionality

### PR DESCRIPTION
**Pull request by ShaBloze**
## Description

Show price search page with search parameters and table with mock data.

## References

- #1655
https://app.zenhub.com/workspaces/team-volt-60a61059028fd7000e49956e/issues/energinet-datahub/geh-charges/1655

### CODE REVIEW

File: libs/dh/charges/domain/src/lib/chargeTypes.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export enum ChargeTypes {
+  Unknown = 'Unknown',
+  Subscription = 'Subscription',
+  Fee = 'Fee',
+  Tariff = 'Tarif',
```
Comment (x-platformcoder) [inline libs/dh/charges/domain/src/lib/chargeTypes.ts:21]:
```suggestion
  Tariff = 'Tariff',
```

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+  private destroy$ = new Subject<void>();
+
+  iconSizes = WattIconSize;
+  searchResult: Array<object> = [
+    {
+      id: 1,
+      name: 'test name',
+      owner: 'test owner',
+      isTax: true,
```
Comment (x-platformcoder) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.ts]: I know this is just static demodata, bug isTax cannot be true on a subscription

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            label="openInNew"
+            [size]="iconSizes.Medium"
+          ></watt-icon>
+          {{ transloco("Download") }}
+        </watt-button>
+
+        <!-- Pagination -->
+        <mat-paginator
+          [pageSizeOptions]="[5, 10, 20, 50]"
+          pageSize="10"
```
Comment (ManBearTM) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:39]: suggestion: use `[50, 100, 150, 200, 250]` and`100` as default according to UX (will probably be refined later).

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+      <!-- icons column -->
+      <ng-container matColumnDef="icons">
+        <th mat-header-cell *matHeaderCellDef>-</th>
+        <td mat-cell *matCellDef="let element">
+          <watt-icon
+            name="meter"
+            label="meter"
+            [size]="iconSizes.Small"
+          ></watt-icon>
```
Comment (ManBearTM) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:77]: issue: according to design there should be two icons with different colors based on row data.

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    this.dataSource.data = this.searchResult;
+    this.dataSource.paginator = this.paginator;
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
+  }
+
+  private readonly setupPaginatiorTranslation = () => {
```
Comment (ManBearTM) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.ts:105]: nitpick: typo, should be `setupPaginatorTranslation`.

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.ts [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  ngOnChanges() {
+    this.dataSource.data = this.searchResult;
+    this.dataSource.paginator = this.paginator;
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
```
Comment (ManBearTM) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.ts:102]:
```suggestion
    this.destroy$.complete();
```

File: libs/dh/globalization/assets-localization/src/assets/i18n/da.json [hunk 7]
```diff
@@ -264,7 +264,54 @@
...
+      "priceName": "Navn",
+      "owner": "Ejer",
+      "chargeType": "Type",
+      "resolution": "Opløsning",
+      "validFromDate": "Gyldig fra",
+      "validToDate": "Gyldig til",
+      "resultLength": "Søgeresultater",
+      "Download": "Download",
+      "noResultsFound": "Intet resultat",
+      "noResultsHelperText": "Prøv at ændre dine søge indstillinger",
```
Comment (ManBearTM) [inline libs/dh/globalization/assets-localization/src/assets/i18n/da.json:292]:
```suggestion
      "noResultsHelperText": "Prøv at ændre dine søgeindstillinger",
```

**Review by ManBearTM — COMMENTED** [review]
Looks good, only some minor things 🙂

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+<ng-container *transloco="let transloco; read: 'charges.prices'">
+  <div class="card-container">
+    <div class="header-container">
+      <h3>{{ this.searchResult.length }} {{ transloco("resultLength") }}</h3>
+
+      <div class="pagination-container">
+        <!-- Download -->
+        <watt-button
+          variant="text"
+          *dhFeatureFlag="'start_wholesale_process_feature_flag'"
```
[the remark this answers is not in the document]
Comment (ShaBloze) [reply libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:26]: No that is a mistake, great spotted. I fixed it.

**Review by ManBearTM — DISMISSED** [review]

**Review by ManBearTM — APPROVED** [review]

**Pull request merged**